provider/pkg/provider/provider.go

```diff
@@ -510,7 +510,7 @@ func (k *googleCloudProvider) Delete(ctx context.Context, req *rpc.DeleteRequest
 	}
 
 	id := req.GetId()
-	uri := fmt.Sprintf("%s/%s", res.BaseUrl, id)
+	uri := fmt.Sprintf("%s%s", res.BaseUrl, id)
```

Comment:
Perhaps just use this?
```suggestion
	uri := path.Join(res.BaseUrl, id)
```

Reply:
It turns out `path.Join` doesn't quite work for URLs with all kinds of weirdness re-encoding the strings. I refactored this to a function on the resource and made some extra slashes trimmed just in case.